provisioner/advanced_tower.yml

```diff
@@ -1,10 +1,71 @@
 ---
 - name: Advanced-Tower workshop_type
-  hosts: localhost
+  hosts: control_nodes
   become: true
   gather_facts: false
-  connection: local
+
   tasks:
-    - name: Print Summary Information
-      debug:
-        msg: "No specific tasks for {{workshop_type}}"
+
+  - name: create SSH key for user {{ username }}
+    user:
+      name: "{{ username }}"
+      generate_ssh_key: true
+
+  - name: slurp the newly created SSH key for later usage
+    slurp:
+      src: /home/{{ username }}/.ssh/id_rsa.pub
+    register: git_authorized_key
+
+  - name: install some additional convenience software on the control node
+    package:
+      name:
+      - tree
+      - git
+      state: latest
+
+  - name: make sure git-shell is in /etc/shells
+    lineinfile:
+      dest: /etc/shells
+      line: /usr/bin/git-shell
+
+  - name: git user exists and uses git-shell
+    user:
+      name: "{{ git_user }}"
+      comment: "Git server user"
+      shell: /usr/bin/git-shell
+
+  - name: add the newly created SSH key to authorized keys
+    authorized_key:
+      user: "{{ git_user }}"
+      state: present
+      key: "{{ git_authorized_key['content'] | b64decode }}"
+
+  # If the directory exists, we don't want to blow away its
+  # contents, because it's probably got valid repo data in it!
+
+  - name: Check Git project directory for existence
```

Comment:
it would probably be good to have a sub directory, in case the student wants to create a second repo

Reply:
It's probably not necessary but then I'll fix also the duplicate comments...